src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ChangeLevelCalculator.kt

```diff
@@ -28,25 +32,19 @@ class ChangeLevelCalculator(
     deliusRegistrations: Collection<Registration>,
     convictions: Collection<Conviction>
   ): TierLevel<ChangeLevel> {
-
-    val orderedRegistrations = deliusRegistrations
-      .filter { it.active }
-      .sortedByDescending { it.startDate }
-
     return when {
-      mandateForChange.hasNoMandate(crn, convictions) -> {
-        TierLevel(ZERO, 0, mapOf(CalculationRule.NO_MANDATE_FOR_CHANGE to 0))
-      }
-      offenderAssessment == null -> {
-        log.info("No valid assessment found for $crn")
-        TierLevel(TWO, 0, mapOf(CalculationRule.NO_VALID_ASSESSMENT to 0))
-      }
+      mandateForChange.hasNoMandate(crn, convictions) -> noMandateTier
+      hasNoAssessment(crn, offenderAssessment) -> noValidAssessmentTier
+
       else -> {
+        val orderedRegistrations = deliusRegistrations
```

Comment:
Why not pass the registrations in and put this logic in getIomNominalPoints as it is only used in there.